src/commands/Minion/groupkill.ts

```diff
@@ -39,28 +39,84 @@ export default class extends BotCommand {
 		return [quantity, duration, perKillTime];
 	}
 
-	checkReqs(users: KlasaUser[], monster: KillableMonster, quantity: number) {
+	async checkReqs(users: KlasaUser[], monster: KillableMonster, quantity: number) {
 		// Check if every user has the requirements for this monster.
+		const numberOfUsers = users.length;
+		let removedReasonStr: string[];
+		let foodId = 0;
+		let amountOfFoodNeeded = 0;
+
 		for (const user of users) {
+			removedReasonStr = [];
+			let checkReqsState = true;
 			if (!user.hasMinion) {
-				throw `${user.username} doesn't have a minion, so they can't join!`;
+				checkReqsState = false;
+				removedReasonStr.push(
+					`${user.username} doesn't have a minion, so they can't join!`
+				);
 			}
 
 			if (user.minionIsBusy) {
-				throw `${user.username} is busy right now and can't join!`;
+				checkReqsState = false;
+				removedReasonStr.push(`${user.username} is busy right now and can't join!`);
 			}
 
 			if (user.isIronman) {
-				throw `${user.username} is an ironman, so they can't join!`;
+				checkReqsState = false;
+				removedReasonStr.push(`${user.username} is an ironman, so they can't join!`);
 			}
 
 			const [hasReqs, reason] = user.hasMonsterRequirements(monster);
 			if (!hasReqs) {
-				throw `${user.username} doesn't have the requirements for this monster: ${reason}`;
+				checkReqsState = false;
+				removedReasonStr.push(
+					`${user.username} doesn't have the requirements for this monster: ${reason}`
+				);
+			}
+
+			if (1 > 2) {
+				let [healAmountNeeded] = calculateMonsterFood(monster, user);
+				healAmountNeeded = Math.ceil(healAmountNeeded / numberOfUsers);
+
+				for (const food of Eatables) {
+					const amountNeeded = ceil(healAmountNeeded / food.healAmount) * quantity;
+					if (user.numItemsInBankSync(food.id) < amountNeeded) {
+						if (Eatables.indexOf(food) === Eatables.length - 1) {
+							checkReqsState = false;
+							removedReasonStr.push(
+								`${user.username} doesn't have enough food to kill ${
+									monster.name
+								} in this sized mass! You need enough food to heal atleast ${healAmountNeeded} HP (${healAmountNeeded /
+									quantity} per kill) You can use these food items: ${Eatables.map(
+									i => i.name
+								).join(', ')}.`
+							);
+						}
+
+						continue;
+					}
+
+					foodId = food.id;
+					amountOfFoodNeeded = amountNeeded;
+					break;
+				}
 			}
 
-			if (1 > 2 && !hasEnoughFoodForMonster(monster, user, quantity)) {
-				throw `${user.username} doesn't have enough food.`;
+			if (!checkReqsState) {
+				users.splice(users.indexOf(user), 1);
+				user.send(removedReasonStr.join(', '));
+			} else {
+				await user.removeItemFromBank(foodId, amountOfFoodNeeded);
```

Comment:
the food removal cant be here because this function is used as a preliminary check for the party leader, meaning they would have food taken out for that check and the final one.  That prelim check could maybe be removed and if the leader fails the final check they can get removed from their own party without crashing it, or just dont move the food removal checks.